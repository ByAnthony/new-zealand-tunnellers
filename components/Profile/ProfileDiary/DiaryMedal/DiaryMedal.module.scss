@use "app/variables.module.scss";

.other-medal {
  font-size: 1rem;

  @include variables.fullwidth-main-card;

  border-bottom: 1px solid variables.$primary-light-color;
  background-color: variables.$primary-dark-color;

  @include variables.tablet {
    font-size: 1.1rem;
  }

  @include variables.desktop {
    font-size: 1.15rem;
  }
}

.citation {
  margin-top: variables.$xxs-margin;
  font-size: 1rem;
  font-weight: 200;
  line-height: 1.6rem;

  @include variables.desktop {
    font-size: 1.1rem;
    line-height: 1.6rem;
  }
}
