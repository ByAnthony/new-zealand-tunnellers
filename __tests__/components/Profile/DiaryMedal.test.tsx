import { render, screen } from "@testing-library/react";
import { DiaryMedal } from "../../../app/components/Profile/ProfileDiary/DiaryMedal/DiaryMedal";
import {
  mockBritishWarMedal,
  mockMedals,
  mockVictoryMedal,
} from "../../mocks/mockTunneller";

const component = <DiaryMedal medals={mockMedals} />;

test("renders the component correctly", () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test("renders British War and Victory Medals correctly", () => {
  render(component);

  expect(screen.getByText("British War Medal")).toBeInTheDocument();
  expect(screen.getAllByRole("img")[0].getAttribute("alt")).toEqual(
    "British War Medal ribbon",
  );
  expect(screen.getAllByRole("img")[0].getAttribute("src")).toEqual(
    "/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=96&q=75",
  );
  expect(screen.getByText("Victory Medal")).toBeInTheDocument();
  expect(screen.getAllByRole("img")[1].getAttribute("alt")).toEqual(
    "Victory Medal ribbon",
  );
  expect(screen.getAllByRole("img")[1].getAttribute("src")).toEqual(
    "/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=96&q=75",
  );
});

test("renders Military Medal correctly", () => {
  const mockMilitaryMedal = {
    name: "Military Medal",
    country: "United Kingdom",
    image: "military-medal.png",
    citation: "For bravery in the field",
  };

  const mockComponent = (
    <DiaryMedal
      medals={[
        { ...mockBritishWarMedal },
        { ...mockVictoryMedal },
        { ...mockMilitaryMedal },
      ]}
    />
  );

  render(mockComponent);

  expect(screen.getByText("British War Medal")).toBeInTheDocument();
  expect(screen.getAllByRole("img")[0].getAttribute("alt")).toEqual(
    "British War Medal ribbon",
  );
  expect(screen.getAllByRole("img")[0].getAttribute("src")).toEqual(
    "/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=96&q=75",
  );
  expect(screen.getByText("Victory Medal")).toBeInTheDocument();
  expect(screen.getAllByRole("img")[1].getAttribute("alt")).toEqual(
    "Victory Medal ribbon",
  );
  expect(screen.getAllByRole("img")[1].getAttribute("src")).toEqual(
    "/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=96&q=75",
  );
  expect(screen.getByText("Military Medal")).toBeInTheDocument();
  expect(screen.getAllByRole("img")[2].getAttribute("alt")).toEqual(
    "Military Medal ribbon",
  );
  expect(screen.getAllByRole("img")[2].getAttribute("src")).toEqual(
    "/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fmilitary-medal.png&w=96&q=75",
  );
  expect(screen.getByText("For bravery in the field.")).toBeInTheDocument();
});

test("renders Foreign Medal with country correctly", () => {
  const mockForeignMedal = {
    name: "Medal for Steadfastness and Loyalty",
    country: "Romania",
    image: "steadfastness-loyalty-medal.png",
    citation:
      "For distinguished services rendered during the course of the campaign",
  };

  const mockComponent = (
    <DiaryMedal
      medals={[
        { ...mockBritishWarMedal },
        { ...mockVictoryMedal },
        { ...mockForeignMedal },
      ]}
    />
  );

  render(mockComponent);

  expect(
    screen.getByText("Medal for Steadfastness and Loyalty (Romania)"),
  ).toBeInTheDocument();
  expect(screen.getAllByRole("img")[2].getAttribute("alt")).toEqual(
    "Medal for Steadfastness and Loyalty ribbon",
  );
  expect(screen.getAllByRole("img")[2].getAttribute("src")).toEqual(
    "/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fsteadfastness-loyalty-medal.png&w=96&q=75",
  );
  expect(
    screen.getByText(
      "For distinguished services rendered during the course of the campaign.",
    ),
  ).toBeInTheDocument();
});

test("does not render the component when medals unknown", () => {
  const mockComponent = <DiaryMedal medals={[]} />;

  const { container } = render(mockComponent);

  expect(container).toBeEmptyDOMElement();
});
