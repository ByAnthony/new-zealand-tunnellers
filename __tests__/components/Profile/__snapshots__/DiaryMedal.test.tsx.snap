// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="halfwidth-cards-container"
  >
    <div
      class="halfwidth-secondary-card"
    >
      <p>
        <img
          alt="British War Medal ribbon"
          data-nimg="1"
          decoding="async"
          height="11"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=96&q=75"
          srcset="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </p>
      <span>
        British War Medal
      </span>
    </div>
    <div
      class="halfwidth-secondary-card"
    >
      <p>
        <img
          alt="Victory Medal ribbon"
          data-nimg="1"
          decoding="async"
          height="11"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=96&q=75"
          srcset="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </p>
      <span>
        Victory Medal
      </span>
    </div>
  </div>
</DocumentFragment>
`;
